hack/lib/build.sh

```diff
@@ -78,20 +81,23 @@ build_binaries() {
       targets=("${YURT_ALL_TARGETS[@]}")
     fi
 
-    local target_bin_dir=$(get_binary_dir_with_arch ${YURT_LOCAL_BIN_DIR})
-    mkdir -p ${target_bin_dir}
-    cd ${target_bin_dir}
-    for binary in "${targets[@]}"; do
-      echo "Building ${binary}"
-      go build -o $(get_output_name $binary) \
-          -ldflags "${goldflags:-}" \
-          -gcflags "${gcflags:-}" ${goflags} $YURT_ROOT/cmd/$(canonicalize_target $binary)
+    for platform in "${SUPPORTED_PLATFORMS[@]}"; do
+      for binary in "${targets[@]}"; do
+           OS=${platform%/*}
+           ARCH=${platform##*/}
+           local target_bin_dir=${YURT_LOCAL_BIN_DIR}/$OS/$ARCH/
+           mkdir -p ${target_bin_dir}
+           cd ${target_bin_dir}
+           echo "Building ${binary} $OS $ARCH"
+           GOOS=$OS GOARCH=$ARCH CGO_ENABLED=0 go build -o $(get_output_name $binary) \
```

Comment:
`release-images.sh` also specifies `go envs` and calls the `build.sh` once per arch. The loop here will result in building redundant binaries. In addition, setting `GOARCH` and other go env here will be in conflict with what `release-images.sh` has set.

Reply:
Please fix this problem.